OCD-3910: ics-relationships endpoints

kekey1 opened the pull request.

kekey1 committed:
feat!: Deprecate /certified_products/{id}/ics_relationships endpoints

[#OCD-3910]

kekey1 committed:
Merge branch 'staging' into OCD-3910

kekey1 committed:
feat!: Add new ics-relationships endpoints for listings

The new endpoints are /certified_products/{chplId}/ics-relationships
(and similar endpoints for the two flavors of CHPL Product Numbers).

[#OCD-3910]

kekey1 committed:
fix!: Clear shared store listings in ICS Family of edited listing

[#OCD-3910]

kekey1 committed:
Merge branch 'staging' into OCD-3910

kekey1 committed:
Merge branch 'staging' into OCD-3910

kekey1 committed:
fix: Stop losing ICS family members while looping

[#OCD-3910]

kekey1 committed:
Merge branch 'staging' into OCD-3910

kekey1 committed:
Merge branch 'staging' into OCD-3910

kekey1 committed:
Merge branch 'staging' into OCD-3910

kekey1 committed:
Merge branch 'staging' into OCD-3910

kekey1 committed:
Merge branch 'staging' into OCD-3910

kekey1 committed:
fix: Use correct ICS terminology in error messages

[#OCD-3910]

tmy1313 commented on chpl/chpl-service/src/main/java/gov/healthit/chpl/listing/ics/ListingIcsNode.java:
```diff
@@ -0,0 +1,54 @@
+        private String chplProductNumber;
+    }
+
+    @Getter
+    @SuperBuilder
+    @NoArgsConstructor
+    @AllArgsConstructor
+    @EqualsAndHashCode
+    public static class IdNamePair implements Serializable {
+        private static final long serialVersionUID = -2377078123832863130L;
+        private Long id;
+        private String name;
+    }
```
Is there a way that we can use the IdNamePairSearchResult here?

kekey1 replied:
Extracted IdNamePair out to a separate domain object and am using it here as well as in all the places there were similar inner classes.

kekey1 committed:
refactor: Extract IdNamePair from multiple places to separate domain obj

Some version of this inner class was used in multiple search result
objects (change requests, developers, listings) as well as in the new
ics relationships response object. This refactor it into a single place
since it has turned out to be commonly used.

[#OCD-3910]

kekey1 committed:
fix: Remove 12 hour caching policy for ics-relationships endpoints

We postulate that this is not really needed to be cached by the browser
- probably it's fast enough. It was causing confusion after listing
updates if this data was cached by the browser because the listing
details and ICS Graph did not match until after a hard refresh.

[#OCD-3910]

kekey1 committed:
Merge branch 'staging' into OCD-3910

kekey1 committed:
fix: Clear ICS relationships from shared store before listing update

The ICS relationships need to be cleared both before and after
because if an ICS relationship is REMOVED from the listing, it is no
longer available when the aspect/annotation runs and misses being
cleared out. It will still show an ICS relationship in it's listing
details when none exists.

[#OCD-3910]

kekey1 committed:
Merge remote-tracking branch 'upstream/staging' into OCD-3910
